scripts/Muon/GUI/Common/plot_widget/plotting_canvas/plot_color_queue.py

```diff
@@ -4,36 +4,49 @@
 #   NScD Oak Ridge National Laboratory, European Spallation Source,
 #   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
 # SPDX - License - Identifier: GPL - 3.0 +
-
 # See https://docs.python.org/3/library/heapq.html for details on heapq
 import heapq
 
 
 class ColorQueue(object):
-    def __init__(self):
-        self._queue = []
-        self._max_index = 0
+    def __init__(self, color_list):
+        """Create a ColorQueue based on an input color_list
+        The entry at index 0 has highest priority
+        color_list could be an input list of color codes
+        e.g ['C0', 'C1', 'C2']
+        or a list of hex colors
+        e.g [#000000,#FF0000....]
+        or rgb values
+        e.g [(0, 0, 0), (255, 0, 0)..]
+        All the above are supported by matplotlib
+        """
+        self._color_cache = {color_list[i]: i for i in range(len(color_list))}
```

Comment:
```suggestion
        self._color_cache = {color_list[colour]: colour for colour in range(len(color_list))}
```

I cant bring myself to type colour without a u

Reply:
This should be I as its an index, rather than the actual color, the color is given by color_list[i]

Reply:
does that mean that `i` is the priority then?

Reply:
It does yeah, I could rename it to priority, but then color_list[priority] isn't so clear either.

Reply:
Another option is to add a comment of `self._colour_cache = {colour: priority}` that should make it clearer. Then we can see that it is getting the colours and priorities. The last option is to have seperate lists, but that seems silly.

Reply:
yeah ill add a comment to its clear